server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java

```diff
@@ -1857,8 +1857,13 @@ private void validateRootVolumeDetachAttach(VolumeVO volume, UserVmVO vm) {
         }
     }
 
-    private Volume orchestrateDetachVolumeFromVM(long vmId, long volumeId) {
+    @ActionEvent(eventType = EventTypes.EVENT_VOLUME_DETACH, eventDescription = "detaching volume")
+    public Volume detachVolumeViaDestroyVM(long vmId, long volumeId) {
+        Volume result = orchestrateDetachVolumeFromVM(vmId, volumeId);
```

Comment:
you can use directly `return  orchestrateDetachVolumeFromVM(vmId, volumeId);`

Reply:
Done